### ISSUE TITLE
Notification occassionally does not respect Do Not Disturb, killing app solves issue

### ISSUE THREAD

**Issue by shombando** [tuskyapp/Tusky#2921]
Tusky normally respects Android's system wide Do Not Disturb (DND) setting and does not play a notification sound. However, this is the third time since I installed the beta that it has started playing notification sounds with the system still on DND. I'm unable to reproduce this issue but killing the app solves the issue. 

I've confirmed that "Override Do Not Disturb" is not enabled for any of the notification categories. I've also confirmed that when the notification playing the audio pops-up it does not show the tiny bell icon which would display on the notification if DND was off. Not sure what other information would be helpful but I'm happy to provide more. I didn't find any instructions on creating a debug log or stack trace but happy to follow instructions. 

* * * *
- Tusky Version: 20.0 beta 1
- Android Version: 13
- Android Device: Pixel 6 (not a, not pro)
- Mastodon instance (if applicable): fosstodon.org

- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.

**Comment by shombando** [comment]
I can confirm that toggling DND off and on does not resolve the issue, neither does swiping it way from the app switcher. "Force Stop" and restating the app fixes the issue.

**Comment by nikclayton** [comment]
From what you've described I think this has to be an Android issue, not a Tusky issue. https://www.google.com/search?q=android+do+not+disturb+notification+sound does show posts from people on different devices reporting similar problems with apps that aren't Tusky, although I'm not sure how much of that is user error.

If DND is on, *and* there are no exceptions, Android should ignore notifications from all apps, and not play a sound. 

Two other things to check:

- Open Android settings, search for "Conversation", check that there are no Tusky related overrides entered there (there shouldn't be, and if I've read the code correctly Tusky doesn't send "Conversation" notifications, but it can't hurt to check)

- Also in Android settings, search for "disturb access", and check that there are no Tusky exceptions there. For example, I see "Google Play Services" and "Google Services Framework" when I look on my device.

**Comment by shombando** [comment]
Hi @nikclayton thanks for the response and troubleshooting tips. I've checked the android related settings (still could be an Android bug). I do not believe it is an exception issue since the issue is intermittent and is resolved by force stopping the app. For the overwhelming majority of the time the DND settings are respected. I've only experienced this twice more since my follow-up report on December 20. I also can't reproduce the issue so I'm happy to close the issue.
